packages/core/lib/std.js

```diff
@@ -1,81 +1,98 @@
-const { resolveFn, resolveFnScope, resolveVar } = require("./scope");
+const { resolveRule, resolveRuleScope, resolveVar } = require("./scope");
 const { help, stringify } = require("./utils");
 const { BrowseError } = require("./error");
 
-const get = (scope) => (_opts) => (name) => resolveVar(name, scope);
-const set = (scope) => (_opts) => (name, value) => {
-  scope.vars[name] = value;
-  return value;
-};
-
 /**
- * The root scope that contains all the basic/standard functions and variables
+ * The root scope that contains all the basic/standard rules and variables
  */
 module.exports = ({ evalRule, evalRuleSet, getNewScope }) => ({
   parent: null, // This is the root
   vars: {},
   internal: {},
-  fns: {
-    help: (scope) => (_opts) => (key) => {
-      // Find the lowest scope that actually has the 'help' function
-      const helpScope = resolveFnScope("help", scope);
+  close: async () => {},
+  rules: {
```

Comment:
I noticed this is done differently in different places. Some places define the rules as an empty object and add to them later. Some places define the rules directly in the object like this. Standardizing how we do that will make for a more readable structure (since we want to OSS) and will also help with more opinionated doc generation. I personally think that

const get = async () => .....

const set = ....

const for = ... 

const if = ....

module.exports = ({ evalRule, .... }) => {
   rules: {
     get,
     set,
     for,
     if
   }
}

might be the cleanest style so we have the option to move individual rules out to files later as the set of rules for a scope gets too large to manage in a mono-file. Thoughts on this??

Reply:
I had to change it to creating a scope object first which has a name, and then creating all the functions so that they can reference that object inside the closure.

so that should be the way to go

well, in realty, the cleaner way here is probably to actually use classes :P Then we can use `this` to modify vars, rules and internal in the scope

right now, I'm trying to use closures as a way for the rules to maintain state, and classes are better here

Happy to chat in person if anything is still unanswered here, and yeah, I wanna make sure whatever format we use for the scopes helps with your doc generation. Maybe I can abstract a utility for scope definition